e2e_tests/tests/task/test_generic_tasks.py

```diff
@@ -0,0 +1,200 @@
+            )
+            result_config = util.yaml_safe_load(res.stdout)
+            expected_config = {"entrypoint": ["python3", "run_fork.py"]}
+            assert expected_config == result_config
+
+
+@pytest.mark.e2e_cpu
+def test_kill_generic_task() -> None:
+    """
+    Start a simple task without a context directory and grab its config
+    """
+    test_session = api_utils.determined_test_session()
+
+    with open(conf.fixtures_path("generic_task/test_config.yaml"), "r") as config_file:
+        # Create task
+        config = ntsc.parse_config(config_file, None, [], [])
+        config_text = util.yaml_safe_dump(config)
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=None,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+        # Kill task
+        command = ["det", "-m", conf.make_master_url(), "task", "kill", task_resp.taskId]
+
+        subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+
+        kill_resp = bindings.get_GetTask(test_session, taskId=task_resp.taskId)
+        assert bindings.v1GenericTaskState.CANCELED == kill_resp.task.taskState
+
+
+@pytest.mark.e2e_cpu
+def test_pause_and_unpause_generic_task() -> None:
+    """
+    Start a simple task without a context directory and grab its config
+    """
+    test_session = api_utils.determined_test_session()
+
+    with open(conf.fixtures_path("generic_task/test_config.yaml"), "r") as config_file:
+        # Create task
+        config = ntsc.parse_config(config_file, None, [], [])
+        config_text = util.yaml_safe_dump(config)
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=None,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+        # Pause task
+        command = ["det", "-m", conf.make_master_url(), "task", "pause", task_resp.taskId]
+
+        subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+
+        pause_resp = bindings.get_GetTask(test_session, taskId=task_resp.taskId)
+        assert bindings.v1GenericTaskState.PAUSED == pause_resp.task.taskState
+
+        # Unpause task
+        command = ["det", "-m", conf.make_master_url(), "task", "unpause", task_resp.taskId]
```

Comment:
Not really a comment on the test, but out of curiosity: what happens inside the generic task if somebody pauses-then-unpauses so quickly?

Reply:
There is a task state check in unpause for a `PAUSED` state. A task is only assigned that state when it's allocation is terminated by a pause command. So in the situation where we unpause before the allocation is terminated the command would return an error due to the task not being in a paused state.

Reply:
so does the task transition to `PAUSED` as soon as the pause command is issued?  Or does it wait for the task to terminate?

Also, does `det task pause` block until the allocation terminates?

Reply:
Correction I had it wrong in the above comment.  Unpause will not be blocked. It will override the pause action. There is a intermediate state between `ACITVE` and `PAUSED` called `STOPPING_PAUSED` that a task will be set to at the beginning of the pause action.